check_no_globals.go

```diff
@@ -18,6 +20,68 @@ type allowedExpression struct {
 	SelName string
 }
 
+// reports is the reports of found global variables that's not valid and will be
+// reported to the analysis pass.
+type report struct {
+	name string
+	pos  token.Pos
+}
+
+// Analyzer is the analasys analyzer for gochecknoglobals. Ironically enough,
+// this is in fact a global variable.
+var Analyzer = &analysis.Analyzer{ //nolint
+	Name:             "gochecknoglobals",
+	Doc:              "Don't allow global variables",
+	Run:              run,
+	Flags:            flags(),
+	Requires:         []*analysis.Analyzer{nolint.Analyzer},
+	RunDespiteErrors: true,
+}
+
+func flags() flag.FlagSet {
+	flags := flag.NewFlagSet("", flag.ExitOnError)
+	flags.Bool("t", false, "Include tests")
+
+	return *flags
+}
+
+func run(pass *analysis.Pass) (interface{}, error) {
+	noLinter := pass.ResultOf[nolint.Analyzer].(*nolint.NoLinter)
+	runTests, _ := strconv.ParseBool(pass.Analyzer.Flags.Lookup("t").Value.String())
```

Comment:
Not really, there's no `Get()` method on `flag.FlagSet`. Lookup returns a `Value` and every built in `Value` implements the `flag.Getter` interface. An alternative way to write this with type casting instead of parsing would be like this:

```go
flag.Lookup("t").Value.(flag.Getter).Get().(bool)
```

You think this is preferred?

Reply:
If that gets the raw bool that the flag package already parsed, I think that is preferred.